gef.py

```diff
@@ -8156,14 +8157,26 @@ def context_trace(self):
         nb_backtrace = self.get_setting("nb_lines_backtrace")
         if nb_backtrace <= 0:
             return
-        orig_frame = current_frame = gdb.selected_frame()
-        i = 0
 
         # backward compat for gdb (gdb < 7.10)
         if not hasattr(gdb, "FrameDecorator"):
             gdb.execute("backtrace {:d}".format(nb_backtrace))
             return
 
+        orig_frame = gdb.selected_frame()
+        current_frame = gdb.newest_frame()
+        frames = []
+        frames.append(current_frame)
+        while current_frame != gdb.selected_frame():
+            if not current_frame.older():
```

Comment:
Why not just put this in the `while` predicate?

Reply:
I just removed this condition because it is never true.